sklearn/gaussian_process/_gpr.py

```diff
@@ -461,7 +461,7 @@ def predict(self, X, return_std=False, return_cov=False):
             else:
                 return y_mean
 
-    def sample_y(self, X, n_samples=1, random_state=0):
+    def sample_y(self, X, n_samples=1, random_state="warn"):
```

Comment:
In the `random_state` docstring:

https://github.com/scikit-learn/scikit-learn/blob/59a48db013b14adbd758385efe2215a3649df47c/sklearn/gaussian_process/_gpr.py#L475-L479

can you add this:

```
        .. versionchanged:: 1.5
           The default value of `random_state` will change to `None` in v1.5.
```